engine/ast.ml

```diff
@@ -1,7 +1,6 @@
 [@@@ warning "-30"]
 
 type source_program  = {
-  scrutinee: variable;
```

Comment:
I got rid of the `scrutinee` field in source progra, due to the fact that many of our examples use `function ...`, which does not name the input parameter. We may want to add it back (as a variable option? generating a fresh/dummy name for `function`?) if it is useful later.

Reply:
Ok